### DISCUSSION TITLE
How to automatically record exception information?

### DISCUSSION THREAD

**Discussion by tuhao1020** [open-telemetry/opentelemetry-java-instrumentation#6382] [Q&A] [answered]
<img width="1250" alt="image" src="https://user-images.githubusercontent.com/26807520/181467994-59b0d3f4-abf3-47fb-9ab6-8427787cdeb3.png">

Exceptions are thrown in a method, but the details of the exceptions, such as message, line of code, etc., are not recorded in span, can they be automatically recorded when using automatic instrumentation?

**Comment by mateuszrzeszutek** [comment, marked as answer]
Hey @tuhao1020 ,
Try using a span exporter other than Zipkin - both Jaeger and OTLP preserve full exception details.
In OpenTelemetry, an exception is recorded as a span event. The equivalent of an event in Zipkin is an annotation; however, Zipkin. annotations cannot have attributes. That's why during conversion to Zipkin any span event attributes (exception message, exception stack trace, etc.) are just dropped.

**Reply by tuhao1020** [reply]
I am using jaeger instead of zipkin in OTEL_TRACES_EXPORTER， but exception stack trace still dropped
```
OTEL_TRACES_EXPORTER: jaeger,logging
OTEL_EXPORTER_JAEGER_ENDPOINT: http://jaeger:14268/api/traces
```

**Reply by changliu-wk** [reply]
@tuhao1020, are you using @WithSpan annotation?

**Reply by tuhao1020** [reply]
Sorry, I forgot to update the status, I used a customized jaeger-ui and the details of the exception actually already existed
